lib/features/wallet_screen/add_contact_page/add_contact_page.dart

```diff
@@ -0,0 +1,145 @@
+import 'dart:async';
+
+import 'package:auto_route/auto_route.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:hive/hive.dart';
+import 'package:threedpass/core/polkawallet/app_service.dart';
+import 'package:threedpass/core/polkawallet/bloc/app_service_cubit.dart';
+import 'package:threedpass/core/utils/show_text_snackbar.dart';
+import 'package:threedpass/core/widgets/buttons/card_elevated_button.dart';
+import 'package:threedpass/core/widgets/d3p_scaffold.dart';
+import 'package:threedpass/core/widgets/input/textformfield/textformfield.dart';
+import 'package:threedpass/core/widgets/paddings.dart';
+import 'package:threedpass/features/wallet_screen/add_contact_page/bloc/contacts_bloc.dart';
+import 'package:threedpass/features/wallet_screen/add_contact_page/domain/entities/contact.dart';
+
+@RoutePage()
+class AddContactPage extends StatelessWidget {
+  AddContactPage({final Key? key}) : super(key: key);
+
+  final _formKey = GlobalKey<FormState>();
+  late final AppService appService;
+  final TextEditingController nameTextController = TextEditingController();
+  final TextEditingController addressTextController = TextEditingController();
+
+  @override
+  Widget build(final BuildContext context) {
+    appService = BlocProvider.of<AppServiceLoaderCubit>(context).state;
+    return D3pScaffold(
+      appbarTitle: 'add_contact_appbar_title',
+      body: SafeArea(
```

Comment:
Does SafeArea do a useful job at this point? We have an appbar and navigationBar that protect the main page, right?